packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -855,15 +854,24 @@ class _AndroidDevicePortForwarder extends DevicePortForwarder {
   }
 
   @override
-  Future<int> forward(int devicePort, { int hostPort }) async {
-    if ((hostPort == null) || (hostPort == 0)) {
-      // Auto select host port.
-      hostPort = await portScanner.findAvailablePort();
-    }
-
-    await runCheckedAsync(device.adbCommandForDevice(
+  Future<int> forward(int devicePort, {int hostPort}) async {
+    hostPort ??= 0;
+    final RunResult process = await runCheckedAsync(device.adbCommandForDevice(
```

Comment:
`assert(hostPort >= 0)`?

Reply:
we don't guarantee that. We could audit all the call sites to make sure that the callers all check the range on the port, but as of right now we're depending on adb to do that.